dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java

```diff
@@ -162,6 +169,116 @@ private ResponseEntity<?> scaleBrokers(
     }
   }
 
+  @PatchMapping(consumes = "application/json", produces = "application/json")
+  public ResponseEntity<?> updateClusterConfiguration(
+      @RequestParam(defaultValue = "false") final boolean dryRun,
+      @RequestParam(defaultValue = "false") final boolean force,
+      @RequestBody final ClusterConfigPatchRequest request) {
+    try {
+      final var brokers = request.getBrokers();
+      final var partitions = request.getPartitions();
+      if (force) {
+        return forceRemoveBrokers(dryRun, brokers, partitions);
+      }
+
+      if (brokers != null
+          && brokers.getCount() != null
+          && ((brokers.getAdd() != null || !brokers.getAdd().isEmpty())
+              || (brokers.getRemove() != null || !brokers.getRemove().isEmpty()))) {
```

Comment:
:question: Shouldn't it be `!= null && !isEmpty()`? Otherwise, if one is null we'd next check `isEmpty` which would result in an NPE